### PR TITLE
fix(optimizer): internal tables return no stream key

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- Resolve https://github.com/risingwavelabs/risingwave/issues/14301
- Internal tables have no stream key, so we just return a None for stream_key of internal table. Otherwise, functional dependencies could be derived from the wrong info.

```
dev=> explain select count(*) from __internal_v_4_hashjoinleft_1003 group by t_a;
                                      QUERY PLAN
---------------------------------------------------------------------------------------
 BatchExchange { order: [], dist: Single }
 └─BatchProject { exprs: [count] }
   └─BatchSortAgg { group_key: [__internal_v_4_hashjoinleft_1003.t_a], aggs: [count] }
     └─BatchScan { table: __internal_v_4_hashjoinleft_1003, columns: [t_a] }
(4 rows)
```

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/frontend/src/optimizer/plan_node/generic/scan.rs [hunk 1]
```diff
@@ -343,6 +344,9 @@ impl GenericPlanNode for Scan {
     }
 
     fn stream_key(&self) -> Option<Vec<usize>> {
+        if matches!(self.table_catalog.table_type, TableType::Internal) {
+            return None;
+        }
         let id_to_op_idx = Self::get_id_to_op_idx_mapping(&self.output_col_idx, &self.table_desc);
         self.table_desc
             .stream_key
```
Comment (stdrc) [inline src/frontend/src/optimizer/plan_node/generic/scan.rs:352]: Doesn't the issue mean that the `stream_key` field of `TableDesc` is not reasonable?

Comment (stdrc) [reply src/frontend/src/optimizer/plan_node/generic/scan.rs:352]: It still sounds strange that `TableType::Table` has `stream_key` even we handled `TableType::Internal`...

**Review by st1page — APPROVED** [review]
LGTM as a tentative fix. Maybe we have to choose one later.
1. use the primary key as the stream key, introducing a new invariant that "all tables must have stream key"
2. change the `stream key`'s type in the tableCatalog from `Vec` to `Option<Vec>`

slightly prefer 1.

**Comment by chenzl25** [comment]
I agree the current fix looks like a quick fix. However, all internal tables that persisted previously have `stream_key = vec![]` We can't tell whether the stream key exists from. We can use the primary key for internal tables as the stream key, however, it is like a quick fix as well. For backward compatibility, let's choose the simplest fix first.

**Pull request merged**